.github/workflows/test-and-release.yml

```diff
@@ -0,0 +1,111 @@
+name: Test and Release
+on:
+  push:
+    branches:
+      - main
+      - next
+    tags:
+      - '*'
+  pull_request:
+permissions:
+  contents: write
+jobs:
+  full:
+    name: Node.js Latest Full
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run tests
+        run: pnpm test
+  short:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version:
+          - 18
+          - 16
+    name: Node.js ${{ matrix.node-version }} Quick
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run unit tests
+        run: pnpm unit
+  benchmark:
+    name: Benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install --ignore-scripts
+      - name: Run benchmark
+        run: node ./test/benchmark.js
+  release:
+    name: Release On Tag
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+    needs: [full, short, benchmark]
```

Comment:
Let’s create a release on new tag without waiting for tests.

Anyway tag was created.

Reply:
This will simplify the workflow. Given the previous comment, we'll have a total of three workflow files: test, benchmark (won't trigger on PR), release (won't depend on tests). Is this correct?

Reply:
I think we should have 2 workflow files: `test.yml` and `release.yml`

Reply:
Ah yes, sorry, thought the previous conversation was pointing to the benchmark.

Reply:
We can keep `benchmark` in `test.yml` as we had before (since we need to run it on PR)